packages/component/src/BasicTranscript.js

```diff
@@ -1038,41 +1041,58 @@ const SetScroller = ({ activityElementsRef, scrollerRef }) => {
         typeof autoScrollSnapOnPageOffset === 'number' ? autoScrollSnapOnPageOffset : 0;
 
       if (patchedAutoScrollSnapOnActivity || patchedAutoScrollSnapOnPage) {
+        const { current: activities } = activitiesRef;
+        const { current: activityElements } = activityElementsRef;
         const { current: lastAcknowledgedActivity } = lastAcknowledgedActivityRef;
-
         const values = [];
 
-        if (patchedAutoScrollSnapOnActivity) {
-          const { element: nthUnacknowledgedActivityElement } =
-            activityElementsRef.current[
-              activityElementsRef.current.findIndex(({ activity }) => activity === lastAcknowledgedActivity) +
-                patchedAutoScrollSnapOnActivity
-            ] || {};
-
-          if (nthUnacknowledgedActivityElement) {
-            values.push(
-              nthUnacknowledgedActivityElement.offsetTop +
-                nthUnacknowledgedActivityElement.offsetHeight -
-                offsetHeight -
-                scrollTop +
-                patchedAutoScrollSnapOnActivityOffset
-            );
+        const lastAcknowledgedActivityIndex = activities.indexOf(lastAcknowledgedActivity);
+
+        if (~lastAcknowledgedActivityIndex) {
+          // The activity that we acknowledged could be not rendered, such as post back activity.
+          // When calculating scroll snap, we can only base on the first unacknowledged-and-rendering activity.
+          let firstUnacknowledgedActivityElementIndex = -1;
+
+          for (let index = lastAcknowledgedActivityIndex + 1, { length } = activities; index < length; index++) {
+            const activity = activities[index];
+            const activityElementIndex = activityElements.findIndex(entry => entry.activity === activity);
+
+            if (~activityElementIndex) {
+              firstUnacknowledgedActivityElementIndex = activityElementIndex;
+              break;
+            }
           }
-        }
 
-        if (patchedAutoScrollSnapOnPage) {
-          const { element: firstUnacknowledgedActivityElement } =
-            activityElementsRef.current[
-              activityElementsRef.current.findIndex(({ activity }) => activity === lastAcknowledgedActivity) + 1
-            ] || {};
-
-          if (firstUnacknowledgedActivityElement) {
-            values.push(
-              firstUnacknowledgedActivityElement.offsetTop -
-                scrollTop -
-                offsetHeight * (1 - patchedAutoScrollSnapOnPage) +
-                patchedAutoScrollSnapOnPageOffset
-            );
+          if (~firstUnacknowledgedActivityElementIndex) {
+            if (patchedAutoScrollSnapOnActivity) {
+              // Find the activity element which we should snap to.
```

Comment:
Nit on use of "Find" in the comment. We're not "finding" in the known sense, but instead are just accessing the location where it sits, correct?

Reply:
Right.

```
// Gets the activity element which we should snap to.
```